src/quirks/mod.rs

```diff
@@ -0,0 +1,104 @@
+use http::{header, Method};
+use regex::Regex;
+use reqwest::{Request, Url};
+
+/// Sadly some pages only return plaintext results if Google is trying to crawl them.
+const GOOGLEBOT: &str = "Mozilla/5.0 (compatible; Googlebot/2.1; +http://google.com/bot.html)";
+
+#[derive(Debug, Clone)]
+pub struct Quirk {
+    pub pattern: Regex,
+    pub rewrite: fn(Request) -> Request,
+}
+
+#[derive(Debug, Clone)]
+pub struct Quirks {
+    quirks: Vec<Quirk>,
+}
+
+impl Quirks {
+    pub fn init() -> Self {
+        let quirks = vec![
+            Quirk {
+                pattern: Regex::new(r"^(https?://)?(www\.)?twitter.com").unwrap(),
+                rewrite: |request| {
+                    let mut out = request;
+                    *out.method_mut() = Method::HEAD;
+                    out.headers_mut()
+                        .insert(header::USER_AGENT, GOOGLEBOT.parse().unwrap());
+                    out
+                },
+            },
+            Quirk {
+                // https://stackoverflow.com/a/19377429/270334
+                pattern: Regex::new(r"^(https?://)?(www\.)?(youtube\.com|youtu\.?be)").unwrap(),
+                rewrite: |request| {
+                    let mut out = request;
+                    let original_url = out.url();
+                    let urlencoded: String =
+                        url::form_urlencoded::byte_serialize(original_url.as_str().as_bytes())
+                            .collect();
+                    *out.method_mut() = Method::HEAD;
```

Comment:
It seems that making a GET request works here, while HEAD does not (verified with curl).

Reply:
You're right. Checked on my machine and got the same results. Fixed now and added a unit test for verification. Thanks!